libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/BotFrameworkHttpClient.cs

```diff
@@ -104,6 +109,19 @@ public async Task<InvokeResponse> PostActivityAsync(string fromBotId, string toB
             }
         }
 
+        /// <summary>
+        /// Logic to build an <see cref="AppCreentials"/> object to be used to acquire tokens
+        /// for this HttpClient.
+        /// </summary>
+        /// <param name="appId">The application id.</param>
+        /// <param name="oAuthScope">The optional OAuth scope.</param>
+        /// <returns>The app credentials to be used to acquire tokens.</returns>
+        protected virtual async Task<AppCredentials> BuildCredentialsAsync(string appId, string oAuthScope = null)
```

Comment:
This seems the same as the code in the BotFrameworkAdapter above. Why do we need it in 2 places?

Reply:
The other code is in the new HttpClient which might has no coupling to the adapter at this point, and had a copy of this code. We will unify the code but need some more planning on what the right place is. Might do in a separate, R8 PR so I get exposure of this code from customers.